tests/testthat/test-ignore_step.R

```diff
@@ -0,0 +1,108 @@
+test_that("ignore_step() work correctly", {
+  rec <- recipe(mpg ~ ., data = mtcars)
+
+  rec1234 <- recipe(mpg ~ ., data = mtcars) %>%
+    step_dummy(all_nominal_predictors(), id = "dummy") %>%
+    step_impute_mean(all_numeric_predictors(), id = "impute_mean") %>%
+    step_normalize(all_numeric_predictors(), id = "normalize") %>%
+    step_pca(all_numeric_predictors(), id = "pca")
+
+  rec234 <- recipe(mpg ~ ., data = mtcars) %>%
+    step_impute_mean(all_numeric_predictors(), id = "impute_mean") %>%
+    step_normalize(all_numeric_predictors(), id = "normalize") %>%
+    step_pca(all_numeric_predictors(), id = "pca")
+
+  rec34 <- recipe(mpg ~ ., data = mtcars) %>%
+    step_normalize(all_numeric_predictors(), id = "normalize") %>%
+    step_pca(all_numeric_predictors(), id = "pca")
+
+  rec123 <- recipe(mpg ~ ., data = mtcars) %>%
+    step_dummy(all_nominal_predictors(), id = "dummy") %>%
+    step_impute_mean(all_numeric_predictors(), id = "impute_mean") %>%
+    step_normalize(all_numeric_predictors(), id = "normalize")
+
+  expect_equal(
+    ignore_attr = TRUE,
+    ignore_step(rec1234, number = 1),
+    rec234
+  )
```

Comment:
I'm surprised by the argument order here, why do you put `ignore_attr` up front?